### DISCUSSION TITLE
dwm: combine systray and dwm default bar

### DISCUSSION THREAD

**Discussion by ghost** [siduck/chadwm#191] [Q&A]
![2024-08-02_18-37_1](https://github.com/user-attachments/assets/da4628ab-ba78-49a5-a032-f9bf4378dbab)

Hello siddharth bhaiya,
as you can see, the systray is separated from the bar and is rounded unlike the bar, any ways to fix this? I am using picom and rounded corners

the workaround I do to fix this is, add the window classes under ronded-corners-exclude. 

or putting override_redirect under rounded-corners-exclude. But this approach gives rise to problems like these
![2024-08-02_18-43](https://github.com/user-attachments/assets/4ddc4e59-2bad-4d8a-9abd-92517ec7d6f1)

context menus like these have weird rectangular border shadows.

I have used chadwm a lot, before moving to my own dwm config.

thanks a lot for the lovely project and inspiration.

**Comment by siduck** [comment]
hi @Subhashis2007  ,  i no longer use dwm so cant help, u can try to exclude dwm bar from rounded stuff in picom config i think

**Reply by ghost** [reply]
sadly couldn't get its class name

**Reply by siduck** [reply]
try xprop command and click on dwmbar

**Reply by ghost** [reply]
> try xprop command and click on dwmbar

I had done it, it returns 'dwm', excluded it already, but no luck with systray.

U used picom earlier? I mean with dwm

**Reply by siduck** [reply]
try clicking on systray

**Reply by ghost** [reply]
> try clicking on systray

_NET_SYSTEM_TRAY_ORIENTATION(CARDINAL) = 342

**Reply by siduck** [reply]
idk then, u should ask on suckless tg group or its subreddit

**Reply by ghost** [reply]
added this and it worked, 
_NET_SYSTEM_TRAY_ORIENTATION@:c

**Reply by ghost** [reply]
> added this and it worked, _NET_SYSTEM_TRAY_ORIENTATION@:c

sorry it didn't

**Reply by ghost** [reply]
> > added this and it worked, _NET_SYSTEM_TRAY_ORIENTATION@:c
> 
> sorry it didn't

Imma ask in r/unixporn

**Reply by MinePro120** [reply]
I think this is just the way systray is implemented. If you have any findings though, it would help posting them here.

**Reply by MinePro120** [reply]
The black rounded corners is a picom issue though, I have encountered it before (but don't remember how I solved it). Try changing picom's config.

**Reply by MinePro120** [reply]
An easy workaround for this would (probably) be to modify the systray patch to add some padding on the left side of the systray, so it extends below the bar, thus hiding the rounded corners. I will attempt this in the following days.
